File: app/src/main/java/com/enipro/data/remote/model/Date.java
package com.enipro.data.remote.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import org.joda.time.format.ISODateTimeFormat;
import org.parceler.ParcelConstructor;

import java.text.ParseException;

@org.parceler.Parcel
public class Date {

    @SerializedName("$date")
    @Expose
    public String $date;

    @ParcelConstructor
    public Date(String $date) {
        this.$date = $date;
    }

    public String get_$date() {
        return $date;
    }

    public void set_$date(String $date) {
        this.$date = $date;
    }

    /**
     * Parsed date through a joda date formatter and parsed to give a java.util.Date.
     *
     * @return the java.util.Date object of the date.
     * @throws ParseException
     */
    public org.joda.time.LocalDateTime getUtilDate() {
        org.joda.time.format.DateTimeFormatter parser = ISODateTimeFormat.dateTimeParser();
        return parser.parseDateTime($date).toLocalDateTime();
    }
}

File: app/src/main/java/com/enipro/data/remote/model/Date.kt
package com.enipro.data.remote.model

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import org.joda.time.format.ISODateTimeFormat

data class Date constructor(@field:SerializedName("\$date")
                            @field:Expose
                            var date: String) : Parcelable {
    val utilDate: org.joda.time.LocalDateTime
        get() {
            val parser = ISODateTimeFormat.dateTimeParser()
            return parser.parseDateTime(date).toLocalDateTime()
        }

    constructor(parcel: Parcel) : this(parcel.readString())

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(date)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Date> {
        override fun createFromParcel(parcel: Parcel): Date {
            return Date(parcel)
        }

        override fun newArray(size: Int): Array<Date?> {
            return arrayOfNulls(size)
        }
    }
}
